src/main/java/edu/harvard/iq/dataverse/EditDatafilesPage.java

```diff
@@ -2037,7 +2049,11 @@ public void handleExternalUpload() {
     			// -----------------------------------------------------------
     			if (this.isFileReplaceOperation()){
     				this.handleReplaceFileUpload(storageLocation, fileName, contentType, checksumValue, checksumType);
-    				this.setFileMetadataSelectedForTagsPopup(fileReplacePageHelper.getNewFileMetadatasBeforeSave().get(0));
+                                if (!saveEnabled){
+                                    uploadWarningMessage = fileReplacePageHelper.getErrorMessages();
```

Comment:
From line 1889, isn't uploadWarningMessage already set? And you could just check for non-null on that rather than checking saveEnabled? (It wasn't immediately clear to me why you were checking saveEnabled (in an error case it remains false), but I'm mostly asking because if you do need to set the warning here, I'm not reading the code correctly.)

Reply:
Point taken. I guess I had two things in mind - not calling the "get(0)" if the array was null and making it obvious that the save was not enabled. I can simplify here.

Reply:
Looks good